src/main/java/org/cbioportal/web/columnar/StudyViewColumnStoreController.java

```diff
@@ -213,7 +215,26 @@ public ResponseEntity<List<ClinicalDataCountItem>> fetchClinicalDataCounts(
        // boolean singleStudyUnfiltered = studyViewFilterUtil.isSingleStudyUnfiltered(studyViewFilter);
         List<ClinicalDataCountItem> result = studyViewColumnarService.getClinicalDataCounts(studyViewFilter, 
             attributes.stream().map(ClinicalDataFilter::getAttributeId).collect(Collectors.toList()));
-            //studyIds, sampleIds, attributes.stream().map(a -> a.getAttributeId()).collect(Collectors.toList())); 
+
+        // fetch the samples by using the provided study view filter
+        List<Sample> filteredSamples = studyViewColumnarService.getFilteredSamples(studyViewFilter);
```

Comment:
I would push this business logic down into the service (eventually business logic shouldn't even be at this layer)

and utilize the studyViewRepository.getFilteredSamplesCount().

Probably wrap this in a function.